Application/Services/PostService.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Application.Dtos.Blog;
+using Application.Exceptions;
+using Application.Interfaces;
+using AutoMapper;
+using Core.Constants;
+using Core.Entities;
+using Core.Repositories;
+using Mapster;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using System.ComponentModel.DataAnnotations;
+
+namespace Application.Services
+{
+    public class PostService : IPostService
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly UserManager<User> _userManager;
```

Comment:
this is a pretty heavy dependency for your application project. it also forces you to use `IHttpContextAccessor` to access the user info, and all you really use it for is user ID. You can get the user ID from the controller and pass it as an argument where needed. This will also make this service easier to test - instead of having to mock `UserManager` and `IHttpContextAccessor` you can just mock an ID.